src/main/java/net/datafaker/service/WeightedRandomSelector.java

```diff
@@ -0,0 +1,136 @@
+package net.datafaker.service;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+
+/**
+ * A utility class for selecting a random element from a list based on assigned weights.
+ **/
+public record WeightedRandomSelector(Random random) {
+    private static final String WEIGHT_KEY = "weight";
+    private static final String VALUE_KEY = "value";
+
+    public WeightedRandomSelector(Random random) {
+        this.random = random != null ? random : new Random();
+    }
+
+    /**
+     * Returns a weighted random element from the given list, where each element is represented as a Map
+     * containing a weight and the corresponding value.
+     * <p>
+     *
+     * @param items A list of maps, where each map contains:
+     *              - weight: A Double representing the weight of the element, influencing its selection probability.
+     *              - value: The actual element of type T to be randomly selected based on its weight.
+     * @param <T>   The type of the element to be selected from the list. The value associated with the weight can be of any type.
+     * @return A randomly selected element based on its weight.
+     * @throws IllegalArgumentException if:
+     *                                  - the list is null or empty,
+     *                                  - any item in the list is null or empty,
+     *                                  - the item does not contain 'weight' or 'value' keys,
+     *                                  - any weight is null, non-positive, NaN or infinite,
+     *                                  - any values in the list are not unique or null,
+     *                                  - the sum of weights exceeds Double.MAX_VALUE.
+     */
+    public <T> T select(List<Map<String, Object>> items) {
```

Comment:
```suggestion
    public static <T> T select(List<Map<String, Object>> items) {
```
this seems not addressed yet
not sure if we need it public